### PR TITLE
Metadata/Thumbnail sync on publish

### PR DESCRIPTION
**Pull request by currysoup**

### CODE REVIEW

**Comment by currysoup** [comment]
After far too long trying to figure out why the little thumbnail wasn't displaying in MAM i found out it's because unlisted videos don't have publicly visible thumbnails.

File: public/video-ui/src/services/VideosApi.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ export default {
 
   publishVideo: (videoId) => {
     return pandaReqwest({
-      url: '/api/atom/' + videoId + '/publish',
+      url: '/api2/atom/' + videoId + '/publish',
```
Comment (akash1810) [inline public/video-ui/src/services/VideosApi.js]: should this be `atomId` rather than `videoId`?

File: public/video-ui/src/components/VideoAssetAdd/formComponents/VideoAssetUrl.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ import React from 'react';
 export default class VideoAssetUrl extends React.Component {
 
   onUpdateAssetUrl = (e) => {
-    const newVersion = this.props.video.activeVersion === undefined ? 0 : this.props.video.activeVersion + 1;
+    const newVersion = this.props.video.activeVersion === undefined ? 0 : Math.max(...this.props.video.assets.map(x => x.version)) + 1;
```
Comment (akash1810) [inline public/video-ui/src/components/VideoAssetAdd/formComponents/VideoAssetUrl.js]: why are we doing this?

Comment (currysoup) [inline public/video-ui/src/components/VideoAssetAdd/formComponents/VideoAssetUrl.js]: Say you have asset versions 1, 2 and 3 and you roll back the `activeVersion` to 2, then the new version generated would be 3. This causes a version conflict.

File: app/controllers/Api2.scala [hunk 3]
```diff
@@ -10,13 +10,13 @@ import data.JsonConversions._
 import model.commands.CommandExceptions._
 import model.commands._
 import play.api.Configuration
-import util.{YouTubeConfig, AWSConfig}
+import _root_.util.{ YouTubeConfig, AWSConfig}
```
Comment (currysoup) [inline app/controllers/Api2.scala]: 😞  "Helpful", indeed :)

File: public/video-ui/src/components/VideoAssetAdd/formComponents/VideoAssetUrl.js (hunk 2 above)
Comment (akash1810) [inline public/video-ui/src/components/VideoAssetAdd/formComponents/VideoAssetUrl.js]: should this be fixed server side then?

Comment (currysoup) [inline public/video-ui/src/components/VideoAssetAdd/formComponents/VideoAssetUrl.js]: The pattern we've been following is the client `PUT`s a well formed asset/video/whatever and the server either accepts or rejects it.

We could move to a model where we send the data required to generate the mentioned asset/video/whatever but that would be a bit of a break from convention.

File: public/video-ui/src/services/VideosApi.js (hunk 1 above)
Comment (currysoup) [inline public/video-ui/src/services/VideosApi.js]: Will do in another PR.
